NitroxPatcher/Patches/BuilderTool_Patch.cs

```diff
@@ -4,19 +4,69 @@
 using System.Reflection.Emit;
 using Harmony;
 using NitroxClient.GameLogic;
+using NitroxModel.Core;
 using NitroxModel.Helper;
 using UnityEngine;
 
 namespace NitroxPatcher.Patches
 {
+    public class BuilderTool_OnHover_Patch : NitroxPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(BuilderTool);
+        public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("OnHover", BindingFlags.NonPublic | BindingFlags.Instance,
+                                                                                 null, new[] { typeof(Constructable) }, null);
+
+        public static bool Prefix(BuilderTool __instance, bool __result, Constructable constructable)
+        {
+            bool _result = true;
+
+            _result = NitroxServiceLocator.LocateService<Crafting>().BuilderTool_Pre_OnHover(__instance.gameObject, constructable);
+            if (_result == false)
+            {
+                return _result;
+            }
+            //Add more checks here later for other Objects
+
+            return _result;
+        }
+
+        public static void Postfix(BuilderTool __instance, Constructable constructable)
+        {
+            if (constructable.techType == TechType.Fabricator || constructable.techType == TechType.Workbench)
+            {
+                NitroxServiceLocator.LocateService<Crafting>().BuilderTool_Post_OnHover(__instance.gameObject, constructable);
+            }
+        }
+
+        public override void Patch(HarmonyInstance harmony)
+        {
+            PatchPostfix(harmony, TARGET_METHOD);
+        }
+    }
+
+
     public class BuilderTool_HandleInput_Patch : NitroxPatch
```

Comment:
Our pattern is a single file per method patch.  This makes it easy for people to see if their given method has already been patched before.

Reply:
On investigating the crafter logic i've notived that correctly following, which event is when triggered it is necessary to jump through many different files. And taking over logic from other classes at generating new method patches also needs to jump to other files. The other point is, that for a logic process not only one patched method of Subnautica class is needed rather all of them, to see how which method is modified and which event it triggers at which logic class.